conn.go

```diff
@@ -91,16 +92,18 @@ func (l *chanListener) Addr() net.Addr {
 
 func (l *chanListener) Close() error {
 	l.closeOnce.Do(func() {
-		close(l.conns)
+		close(l.closed)
 	})
 	return nil
 }
 
 func (l *chanListener) Accept() (net.Conn, error) {
-	conn, ok := <-l.conns
-	if !ok {
+	select {
+	case <-l.closed:
 		return nil, errors.New("listener closed")
+	default:
 	}
+	conn := <-l.conns
 	return conn, nil
```

Comment:
```suggestion
	select {
	case <-l.closed:
		return nil, errors.New("listener closed")
	case conn := <-l.conns:
		return conn, nil
	}
```

It used to be that `<-l.conns` would return (nil, false) when the listener was closed. Because that channel is no longer closed, we'll need to select on `l.closed` whenever we try to receive from `l.conns` to prevent blocking indefinitely.

Reply:
Right, there may be a race if not put inside the same `select`.